dart/lib/src/scope.dart

```diff
@@ -13,6 +14,14 @@ class Scope {
   /// for example, the route name: `"/users/<username>/"`.
   String? transaction;
```

Comment:
if a span is instance of the type transaction and bound to the scope, change this field would also change the name of transaction, whats about transaction.name immutability?